src/utils/OpenGLBinding.cpp

```diff
@@ -14,20 +14,130 @@
 
 #include "utils/BackendBinding.h"
 
+#include "common/Assert.h"
 #include "common/Platform.h"
+#include "nxt/nxt_wsi.h"
 
+#include <cstdio>
+#include "glad/glad.h"
 #include "GLFW/glfw3.h"
 
 namespace backend {
     namespace opengl {
         void Init(void* (*getProc)(const char*), nxtProcTable* procs, nxtDevice* device);
-        void HACKCLEAR(nxtDevice device);
-        void InitBackbuffer(nxtDevice device);
-        void CommitBackbuffer(nxtDevice device);
     }
 }
 
 namespace utils {
+    class SwapChainImplGL {
+        public:
+            static nxtSwapChainImplementation Create(GLFWwindow* window) {
+                nxtSwapChainImplementation impl = {};
+                impl.Init = Init;
+                impl.Destroy = Destroy;
+                impl.Configure = Configure;
+                impl.GetNextTexture = GetNextTexture;
+                impl.Present = Present;
+                impl.userData = new SwapChainImplGL(window);
+                return impl;
+            }
+
+        private:
+            GLFWwindow* window = nullptr;
+            uint32_t cfgWidth = 0;
+            uint32_t cfgHeight = 0;
+            GLuint backFBO = 0;
+            GLuint backTexture = 0;
+
+            SwapChainImplGL(GLFWwindow* window)
+                : window(window) {
+            }
+
+            ~SwapChainImplGL() {
+                glDeleteTextures(1, &backTexture);
+                glDeleteFramebuffers(1, &backFBO);
+            }
+
+            void HACKCLEAR() {
```

Comment:
Same thing as for the Metal swapchain, let's remove hackClear and do it in BeginSubpass instead.